lib/codegen/meta-python/base/legalize.py

```diff
@@ -382,6 +383,29 @@
             a << bxor(x, y)
         ))
 
+# Expand bitrev
+# Adapted from Stack Overflow.
+# https://stackoverflow.com/questions/746171/most-efficient-algorithm-for-bit-reversal-from-msb-lsb-to-lsb-msb-in-c
+expand.legalize(
+        a << bitrev32(x),
+        Rtl(
+            b << imul_imm(x, imm64(0x0802)),
+            b1 << band_imm(b, imm64(0x22110)),
+            c << imul_imm(x, imm64(0x8020)),
+            c1 << band_imm(c, imm64(0x88440)),
+            c2 << bor(b1, c1),
+            d << band_imm(c2, imm64(0x10101)),
+            a << ushr_imm(d, imm64(16)),
+        ))
+
+expand.legalize(
+        a << bitrev64(x),
+        Rtl(
+            a1 << imul_imm(x, imm64(0x0202020202)),
+            a2 << band_imm(a1, imm64(0x010884422010)),
+            a << urem_imm(a2, imm64(1023)),
```

Comment:
It looks like the StackOverflow article presents these in a confusing way. I think the algorithms in the patch here are meant to be 8-bit `reverseBits` implementations, for 32-bit and 64-bit processors, which I think is meant to be in the code that follows to implement a reverse for a wider integer.

I recommend looking at the `reverse` version under the heading "Low Memory". That seems to be a pretty good approach, and we can generalize it to 64-bit as well as 16-bit and 8-bit.

And if we want to further specialize for x86, we can specialize that version to use the `bswap` instruction to replace the parts that are swapping bytes. But we can figure that part out later.